<?xml version="1.0"?>
<launch>
		
  <!-- Start RPLIDAR sensor node which provides LaserScan data  -->
  <node name="rplidarNode" pkg="rplidar_ros" type="rplidarNode" output="screen">
    <param name="serial_port" type="string" value="/dev/ttyUSB1"/>
    <param name="serial_baudrate" type="int" value="115200"/>
    <param name="frame_id" type="string" value="laser"/>
    <param name="inverted" type="bool" value="false"/>
    <param name="angle_compensate" type="bool" value="true"/>
    <param name="time_increment" type="int" value="0"/>
  </node>
  
  <!-- User Interface -->
  <node name="user_interface" pkg="user_interface" type="main.py" />
  
  <!-- Send waypoints to UART node based on user input from User Interface node -->
  <node name="send_waypoints" pkg="gophr_hardware" type="send_waypoints.py" />
  
  <!-- Text Detection2 -->
  <!--<node name="text_detection" pkg="gophr_hardware" type="text_detection2.py" />-->
  
  <!-- Servo Lock -->
  <node name="servo" pkg="gophr_hardware" type="servo_lock.py" />
  
  <!-- E-Stop -->
  <node name="estop" pkg="gophr_hardware" type="estop.py" />
  
  <!-- UART: Sends signal to stop robot and when needed also sends waypoint data -->
  <node name="uart" pkg="gophr_hardware" type="uart_pi_hero.py" />
  
  <!-- Obstacle Detetion -->
  <node name="obstacle_detection" pkg="obstacle_detection" type="obstacle_read.py" />
  
    
</launch>
